### DISCUSSION TITLE
Saturations in NCG

### DISCUSSION THREAD

**Discussion by marinsiebert** [idaholab/moose#26624] [Q&A Modules: Porous Flow]
Hello, 

is there a way to output the saturations of the components resulting from calculations of the NCG fluid state? 
So far I have only been able to see saturations for each phase (liquid, gas), however I would like to be able to differentiate between NCG and vapor - as I understand it, the PorousflowPropertyAux does only account for phases when calculating the saturations. Are there any other possiblities to do this?

**Comment by GiudGiud** [comment]
@1runer

**Comment by 1runer** [comment]
PorousflowPropertyAux just allows for outputting the phase saturations or each mass fraction variable in each phase. 
I don't think there are component saturations within each phase calculated. 

Maybe [this page](https://mooseframework.inl.gov/modules/porous_flow/compositional_flash.html) is helpful.
Otherwise @cpgr could probably correct me.

**Reply by cpgr** [reply]
@1runer is correct. You can output the mass fraction of each component in the gas phase using PorousFlowPropertyAux. If you want to compute the fraction of gas phase saturation of each gas component, you could probably do so knowing the mass fractions and densities?

**Comment by marinsiebert** [comment]
Thank you,  I will look into it.
